app/models/vm_or_template/operations.rb

```diff
@@ -93,6 +95,25 @@ def validate_vm_control
     nil
   end
 
+  included do
+    supports :control do
+      msg = if vm.retired?
+              'The VM is retired'
```

Comment:
@mzazrivec : I was thinking that we might need to just mark this here and do the translation in the caller unless the caller is the API. But I did not look into the details.